### PR TITLE
Allow setting ipv4.dhcp-client-id parameter

### PR DESCRIPTION
**Pull request by opoplawski**
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

nmcli
##### Summary:

Allow setting ipv4.dhcp-client-id parameter for an interface.  Fixes #1523
##### Example:

```
- name: Configure dhcp-client-id for CORA network
  nmcli: state=present type=ethernet conn_name=CORA dhcp_client_id={{ ansible_fqdn }}
  tags: network
```

### CODE REVIEW

**Comment by opoplawski** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/1840

File: lib/ansible/modules/network/nmcli.py
Comment (gundalow) [inline lib/ansible/modules/network/nmcli.py]: 2.3

**Comment by gundalow** [comment]
@alcamie101 I believe you are the current maintainer of this module, could you please review. Thanks

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Comment by opoplawski** [comment]
Updated version.

**Review by gundalow — APPROVED** [review]
Still need the maintainer to look at this

**Comment by gundalow** [comment]
@alcamie101 Could you please have a look at this PR. Thanks

**Comment by gundalow** [comment]
Hi,
A number of files have been git moved from `modules/network` to `modules/net_tools`, which has effected this PR. Please rebase this PR to pull in that change.

**Comment by gundalow** [comment]
@opoplawski Could you please rebase this, which will move it to `modules/net_tools` thanks in advance.

**Comment by calfonso** [comment]
@opoplawski still waiting on this one to be rebased and moved into net_tools. Also waiting for the maintainer to respond to the review request. cc alcamie101

**Comment by abadger** [comment]
CC @alcamie101 (ping for review of a parameter addition to the nmcli ansible module)

**Comment by Akasurde** [comment]
@opoplawski Are you still working on this ? If not, let me know, I will take over this ?

**Comment by opoplawski** [comment]
@Akasurde - If you could take this over, that would be great.  Thanks.

**Comment by Akasurde** [comment]
@opoplawski Thanks.

Superseded by #34268

**Pull request closed without merging**
